xbmc/addons/AddonManager.cpp

```diff
@@ -210,22 +210,41 @@ bool CAddonMgr::ReloadSettings(const std::string &id)
 }
 
 VECADDONS CAddonMgr::GetAvailableUpdates() const
+{
+  return GetAvailableUpdatesOrOutdatedAddons(false);
+}
+
+std::vector<std::shared_ptr<IAddon>> CAddonMgr::GetOutdatedAddons() const
+{
+  return GetAvailableUpdatesOrOutdatedAddons(true);
+}
+
+std::vector<std::shared_ptr<IAddon>> CAddonMgr::GetAvailableUpdatesOrOutdatedAddons(
+    bool returnOutdatedAddons) const
 {
   CSingleLock lock(m_critSection);
   auto start = XbmcThreads::SystemClockMillis();
 
-  VECADDONS updates;
+  VECADDONS result;
```

Comment:
As you are refactoring you could change VECADDONS for it's underlying type if you want. Just a suggestion.